server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/MailDelivrer.java

```diff
@@ -147,6 +158,14 @@ private ExecutionResult doDeliver(Mail mail, InternetAddress[] addr, Iterator<Ho
         return ExecutionResult.temporaryFailure();
     }
 
+    private Optional<Collection<InternetAddress>> listDeliveredAddresses(SendFailedException sfe) {
```

Comment:
I think just return `Collection<InternetAddress>` is way simpler

Reply:
agree